### PR TITLE
[LiveChatV2] SessionInit and ChatToken Endpoints

### PR DESCRIPTION
**Pull request by aidhage**
1. Adding new Endpoints for LiveChatv2
2. Adding condition in SDK.ts to check the version and select endpoint accordingly.

TODO: 
Will add unittests for it.

### CODE REVIEW

File: src/SDK.ts [hunk 1]
```diff
@@ -92,6 +96,9 @@ export default class SDK implements ISDK {
     const response = await axiosInstance.get(endpoint);
     const elapsedTimeInMilliseconds = timer.milliSecondsElapsed;
     const { data } = response;
+    if(data.LiveChatVersion && data.LiveChatVersion == LiveChatVersion.V2){
```
Comment (xTEddie) [inline src/SDK.ts:99]: Please use `===` instead of `==`.

File: src/SDK.ts [hunk 2]
```diff
@@ -387,14 +400,20 @@ export default class SDK implements ISDK {
...
-    let endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatSessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
+    let endpoint;
     const axiosInstance = axios.create();
     axiosRetry(axiosInstance, { retries: this.configuration.maxRequestRetriesOnFailure });
 
     const { reconnectId, authenticatedUserToken, initContext, getContext } = sessionInitOptionalParams;
 
     const headers: StringMap = Constants.defaultHeaders;
 
+    if(this.liveChatVersion == LiveChatVersion.V2){
```
Comment (xTEddie) [inline src/SDK.ts:411]: Use `==` instead of `===`

File: src/SDK.ts [hunk 3]
```diff
@@ -252,7 +265,7 @@ export default class SDK implements ISDK {
 
       // Retries until it reaches its limit
       setTimeout(() => {
-        this.getChatToken(requestId, getChatTokenOptionalParams, currentRetryCount + 1).then((response) => resolve(response)).catch((error) => reject(error));
+        this.getChatToken(requestId,getChatTokenOptionalParams, currentRetryCount + 1).then((response) => resolve(response)).catch((error) => reject(error));
```
Comment (xTEddie) [inline src/SDK.ts:268]: nit: space after `requestId,`

File: src/SDK.ts [hunk 4]
```diff
@@ -190,13 +197,19 @@ export default class SDK implements ISDK {
     if (!requestId) {
       requestId = uuidv4();
     }
-    let endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatGetChatTokenPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
-    const headers: StringMap = Constants.defaultHeaders;
+    let endpoint;
+    const headers: StringMap = Constants.defaultHeaders;  
+
+    if(this.liveChatVersion==LiveChatVersion.V2){
```
Comment (xTEddie) [inline src/SDK.ts:203]: Use `==` instead of `===`

nit: space before and after `===`

File: src/SDK.ts [hunk 5]
```diff
@@ -190,13 +197,19 @@ export default class SDK implements ISDK {
...
+      endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatv2GetChatTokenPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
+    }
+    else{
```
Comment (xTEddie) [inline src/SDK.ts:206]: nit: space after `else`

File: src/SDK.ts (hunk 2 above)
Comment (xTEddie) [reply src/SDK.ts:411]: nit: space after `if` and before `{`

File: src/SDK.ts [hunk 6]
```diff
@@ -387,14 +400,20 @@ export default class SDK implements ISDK {
...
     axiosRetry(axiosInstance, { retries: this.configuration.maxRequestRetriesOnFailure });
 
     const { reconnectId, authenticatedUserToken, initContext, getContext } = sessionInitOptionalParams;
 
     const headers: StringMap = Constants.defaultHeaders;
 
+    if(this.liveChatVersion == LiveChatVersion.V2){
+      endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatv2SessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
+    }
+    else{
```
Comment (xTEddie) [inline src/SDK.ts:414]: nit: space after `else`

File: src/SDK.ts [hunk 7]
```diff
@@ -387,14 +400,20 @@ export default class SDK implements ISDK {
...
+    let endpoint;
     const axiosInstance = axios.create();
     axiosRetry(axiosInstance, { retries: this.configuration.maxRequestRetriesOnFailure });
 
     const { reconnectId, authenticatedUserToken, initContext, getContext } = sessionInitOptionalParams;
 
     const headers: StringMap = Constants.defaultHeaders;
 
+    if(this.liveChatVersion == LiveChatVersion.V2){
+      endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatv2SessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
```
Comment (xTEddie) [inline src/SDK.ts:412]: I'd avoid having an `else` as follow

```ts
let endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatSessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`; // use default endpoint

if (this.liveChatVersion === LiveChatVersion.V2) { // update endpoint only if it's LiveChatV2
    endpoint =`${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatv2SessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
}
```

File: src/SDK.ts (hunk 1 above)
Comment (xTEddie) [reply src/SDK.ts:99]: nit: space after `if` and before `{`

File: src/SDK.ts (hunk 4 above)
Comment (xTEddie) [reply src/SDK.ts:203]: nit: space after `if` and before `{`

Comment (xTEddie) [reply src/SDK.ts:203]: I'd avoid having an `else` as follow

```ts
let endpoint = `${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatGetChatTokenPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`; // use default endpoint

if (this.liveChatVersion === LiveChatVersion.V2) { // update endpoint only if it's LiveChatV2
    endpoint =`${this.omnichannelConfiguration.orgUrl}/${OmnichannelEndpoints.LiveChatv2GetChatTokenPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
}
```

File: src/SDK.ts (hunk 7 above)
Comment (aidhage) [reply src/SDK.ts:409]: done, added endpoint for LiveChat1 as default

File: src/SDK.ts (hunk 6 above)
Comment (aidhage) [reply src/SDK.ts]: removed else

File: src/SDK.ts (hunk 5 above)
Comment (aidhage) [reply src/SDK.ts]: removed else

File: src/SDK.ts (hunk 4 above)
Comment (aidhage) [reply src/SDK.ts]: replace "==" with "===" and added spacing

File: src/SDK.ts (hunk 1 above)
Comment (aidhage) [reply src/SDK.ts:99]: done

File: src/SDK.ts (hunk 3 above)
Comment (aidhage) [reply src/SDK.ts]: added space

File: src/SDK.ts (hunk 2 above)
Comment (aidhage) [reply src/SDK.ts]: done. Added spaces and update "==" with "==="

File: src/SDK.ts (hunk 1 above)
Comment (xTEddie) [inline src/SDK.ts:99]: Use `===` instead of `==`

nit: space after `if` and before `{`

Comment (aidhage) [reply src/SDK.ts]: Done

**Review by xTEddie — APPROVED** [review]

**Pull request merged**